evap/staff/views.py

```diff
@@ -894,6 +894,10 @@ def make_questionnaire_edit_forms(request, questionnaire, editable):
 
     if not editable:
         editable_fields = ['staff_only', 'obsolete', 'name_de', 'name_en', 'description_de', 'description_en']
+        if questionnaire.type != Questionnaire.CONTRIBUTOR:
+            editable_fields.append('type')
+            form.disallow_type_change_to_contributor = True
+
```

Comment:
can you check whether we can restrict the choices of the type field? that would have the advantage that the user sees only the valid choices, and maybe the validation is also done for us.